### PR TITLE
Work around for hard coded flavor.

### PR DESCRIPTION
**Pull request by ghost**
Work around for the hard coded '1' for showing server details. New VS will show the correct flavor id.

### CODE REVIEW

**Comment by ghost** [comment]
New commit title (pep8 doesn't like the period)

**Comment by jimlindeman** [comment]
+1
Looks good, don't think we can ever handle the case for previously-generated instances so existing hardcode should be fine.

File: jumpgate/compute/drivers/sl/servers.py [hunk 1]
```diff
@@ -445,9 +455,26 @@ def get_server_details_dict(app, req, instance):
...
+            try:
+                import re
+                # Try to retrieve the flavor id in tags
+                flavor_id = int(re.search(r'\d+', tag['tag']['name']).group())
+                flavor_url = app.get_endpoint_url(
+                    'compute', req, 'v2_flavor', flavor_id=flavor_id)
+            except Exception:
+                pass
+
+    # TODO(kmcdonald) - Work around for hardcoded ID for old VS
```
Comment (sudorandom) [inline jumpgate/compute/drivers/sl/servers.py]: Hey, I didn't write this! Be sure to change the author for the note if you make a new one.

File: jumpgate/compute/drivers/sl/servers.py [hunk 2]
```diff
@@ -445,9 +455,26 @@ def get_server_details_dict(app, req, instance):
...
-        'compute', req, 'v2_flavor', flavor_id=1)
+    client = req.env['sl_client']
+    vs = client['Virtual_Guest']
+    tags = vs.getTagReferences(id=instance['id'])
+    flavor_url = None
+    flavor_id = 1
+    for tag in tags:
+        if 'flavor_id' in tag['tag']['name']:
+            try:
+                import re
```
Comment (sudorandom) [inline jumpgate/compute/drivers/sl/servers.py]: Imports should all be at the top except for very specific cases... I don't think this is one of those cases.

File: jumpgate/compute/drivers/sl/servers.py [hunk 3]
```diff
@@ -445,9 +455,26 @@ def get_server_details_dict(app, req, instance):
...
     tenant_id = instance['accountId']
 
-    # TODO(kmcdonald) - Don't hardcode this flavor ID
-    flavor_url = app.get_endpoint_url(
-        'compute', req, 'v2_flavor', flavor_id=1)
+    client = req.env['sl_client']
+    vs = client['Virtual_Guest']
+    tags = vs.getTagReferences(id=instance['id'])
+    flavor_url = None
+    flavor_id = 1
```
Comment (sudorandom) [inline jumpgate/compute/drivers/sl/servers.py:466]: As the code is right now, this line is not needed.

Comment (ghost) [inline jumpgate/compute/drivers/sl/servers.py:466]: This line is needed for line 544 if the VS does not have a flavor_id tag.

**Comment by ghost** [comment]
Fixed the above comments + fixed optimization issue when listing all the virtual servers.

File: jumpgate/compute/drivers/sl/servers.py (hunk 3 above)
Comment (sudorandom) [inline jumpgate/compute/drivers/sl/servers.py:466]: Sorry, my mistake

**Comment by sudorandom** [comment]
+1

**Pull request merged**